### DISCUSSION TITLE
Support for Bridging ERC-20 on OP Stack chains

### DISCUSSION THREAD

**Discussion by albertlai431** [wevm/viem#3049] [Idea / Feature Request]
To reach feature parity with optimism-sdk (https://docs.optimism.io/builders/app-developers/tutorials/cross-dom-bridge-erc20), viem should support deposits and withdrawals of ERC-20 tokens

Discord discussion: https://discord.com/channels/1156791276818157609/1156791519089541241/1303035693349601310

**Comment by gndelia** [comment]
I implemented the depositing/withdrawing of ETH/ERC20 tokens on a fork of OP or a webapp.

Using the Portal contract did not work, at least with the old version of OP. So I had to use a custom implementation for depositing and initiating withdrawals for ERC20, using the L1StandardBridge and the L2Bridge. In case anyone needs it, [here is the deposit](https://github.com/hemilabs/ui-monorepo/blob/05e51c5965e70f72993332945c357ba1746ec5f2/packages/hemi-tunnel-actions/src/depositErc20.ts) and the [withdrawal](https://github.com/hemilabs/ui-monorepo/blob/main/packages/hemi-tunnel-actions/src/initiateWithdraw.ts).

We may eventually move this into a publishable package
